charts/zenko-zookeeper/templates/statefulset.yaml

```diff
@@ -43,7 +43,13 @@ spec:
         - name : ZOO_REPLICAS
           value: "{{ .Values.servers }}"
         - name : ZOO_HEAP_SIZE
-          value: "{{ .Values.heap }}"
+        {{- if .Values.resources.limits.memory }}
```

Comment:
Should be the other way around: you want to use the *requested* amount of memory by default. If that's not given, you want to use the *limit* amount (internally in Kubernetes request = limit when not set), then fallback to 2G.

Also, instead of looking up the value from `.Values`, there may be a way to use a `valueRef` instead, maybe even removing the need to switch-case here.

Reply:
Yeah you're right I'll switch around the order. Also I couldn't really find a cleaner way to do a switch case but I'll checkout valueRef

Reply:
See https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/